go/nbs/table_persister.go

```diff
@@ -69,26 +69,21 @@ func (csbc chunkSourcesByAscendingCount) Less(i, j int) bool {
 }
 func (csbc chunkSourcesByAscendingCount) Swap(i, j int) { csbc[i], csbc[j] = csbc[j], csbc[i] }
 
-type chunkSourcesByDescendingDataSize []sourceWithSize
+type chunkSourcesByDescendingDataSize []chunkSource
 
 func (csbds chunkSourcesByDescendingDataSize) Len() int { return len(csbds) }
 func (csbds chunkSourcesByDescendingDataSize) Less(i, j int) bool {
-	swsI, swsJ := csbds[i], csbds[j]
-	if swsI.dataLen == swsJ.dataLen {
-		hi, hj := swsI.source.hash(), swsJ.source.hash()
+	si, sj := csbds[i], csbds[j]
+	if si.chunkDataLen() == sj.chunkDataLen() {
+		hi, hj := si.hash(), sj.hash()
 		return bytes.Compare(hi[:], hj[:]) < 0
 	}
-	return swsI.dataLen > swsJ.dataLen
+	return si.chunkDataLen() > sj.chunkDataLen()
 }
 func (csbds chunkSourcesByDescendingDataSize) Swap(i, j int) { csbds[i], csbds[j] = csbds[j], csbds[i] }
 
-type sourceWithSize struct {
-	source  chunkSource
-	dataLen uint64
-}
-
 type compactionPlan struct {
-	sources             chunkSourcesByDescendingDataSize
+	sources             chunkSources
```

Comment:
What was the issue with having the type of this field express the requirement that this set of chunkSources by sorted in a specific way?